webui/react/src/hooks/useModal/useModal.tsx

```diff
@@ -51,18 +51,28 @@ const useModal = (config: {
   const prevModalProps = usePrevious(modalProps, undefined);
   const [ modal, contextHolder ] = Modal.useModal();
 
+  /**
+   * The code to close the antd modal is separated out from the code that
+   * calls the `onClose` handler to distinguish the motivation for closing.
+   * `internalClose` is directly used when dealing with closing due to unmounting.
+   * `modalClose` is used when we want to call `onClose` handler if applicable.
+   */
+  const internalClose = useCallback(() => {
+    if (!modalRef.current) return;
+    modalRef.current.destroy();
+    modalRef.current = undefined;
+  }, []);
+
   const modalOpen = useCallback((props: ModalFuncProps = {}) => {
     setModalProps(props);
   }, []);
 
-  const modalClose = useCallback((reason?: ModalCloseReason) => {
-    if (!modalRef.current) return;
-    modalRef.current.destroy();
-    modalRef.current = undefined;
-    if (reason) config.onClose?.(reason);
```

Comment:
if a reason is not provided, the `onClose` event handler that is passed in doesn't get called